Add 'Debug' option for API requests in Apple News to output debugging info

attackant opened the pull request:
A 'Debug' link has been added to the API request in the admin section of the Apple News plugin. This link, which opens in a new tab, can be used for debugging purposes. It is particularly useful for showing the API request.

Also syncs WP scripts node dependencies with WordPress 6.5

Addresses #1110

attackant committed:
Add 'Debug' link to API request in Apple News plugin

A 'Debug' link has been added to the API request in the admin section of the Apple News plugin. This link, which opens in a new tab, can be used for debugging purposes. It is particularly useful for showing the API request.

attackant committed:
Update debug URL in Apple News plugin

The code changes modify the debug URL generation in the Apple News plugin. Instead of directly using the 'get_permalink' method, it now calls the 'action_query_params' method from the Admin_Apple_Index_Page class. This change enhances the URL generation by organizing the query parameters for the debug action.

attackant committed:
Add debug action to Apple News plugin admin

A 'debug' case has been added to the action mapping in the Apple News admin. This action corresponds to a new 'debug_action' function, which initiates a debug process for a post identified by a given ID. This function will help to facilitate issue identification and resolution.

attackant committed:
Refactor push action and add debugging to Apple News plugin

The `push` action code in the Apple News plugin has been significantly refactored adding a debugging method which shows the data that will be sent to the API. The action of getting custom post metadata has been moved into a separate private method, `get_meta`, to keep the class DRY.

attackant committed:
Update Apple News plugin to version 2.4.9

This commit updates the version of the Publish to Apple News plugin from 2.4.8 to 2.4.9. The new version includes debugging for Apple News API requests, as indicated by the change in the readme file. The version number has been updated in all relevant files.

attackant committed:
Updated version and dependencies of 'publish-to-apple-news' plugin

The 'publish-to-apple-news' plugin version has been upgraded from 2.4.8 to 2.4.9. Key dependencies including '@wordpress/*' packages and '@babel/*' packages have also been updated to newer versions, enhancing compatibility with WP 6.5 and resolving potential issues. 'package-lock.json' has been revised to reflect these modifications.

attackant committed:
Fix version number for new base branch Apple News plugin version to 2.5.0

The version of the "Publish to Apple News" WordPress plugin has been updated from 2.4.9 to 2.5.0. This is reflected in all files where the version number is declared. This versioning update indicates the inclusion of new enhancements or bug fixes.

mogmarsh commented on admin/apple-actions/index/class-push.php:
```diff
@@ -540,16 +527,83 @@ private function process_errors( $errors ) {
+		}
+
+		// Get the various settings.
+		$is_paid         = (bool) get_post_meta( $post_id, 'apple_news_is_paid', true );
+		$is_preview      = (bool) get_post_meta( $post_id, 'apple_news_is_preview', true );
+		$is_hidden       = (bool) get_post_meta( $post_id, 'apple_news_is_hidden', true );
+		$is_sponsored    = (bool) get_post_meta( $post_id, 'apple_news_is_sponsored', true );
```
in 2.5.0, these actually can be set by the meta or inherit the channel default, so they won't always be `bool` any more. See https://github.com/alleyinteractive/apple-news/pull/1090/files

attackant replied:
Great, will update.

attackant replied:
Give this a look, thanks Greg.

mogmarsh commented on admin/class-admin-apple-index-page.php:
```diff
@@ -405,6 +407,25 @@ private function push_action( $id ) {
+			return;
+		}
+
+		$action = new Apple_Actions\Index\Push( $this->settings, $id );
```
does this actually push it? Is that what we want to happen?

attackant replied:
No, just calls the constructor and sends in the settings, we only call the `debug` method, not the `push` method, then debug info is printed to the screen and then exits. I'll add a bit of documentation to the README.

attackant replied:
Added some docs here: https://github.com/alleyinteractive/apple-news/wiki/Usage#debug-api-requests

mogmarsh reviewed (changes requested):
This looks good, but there is a change to the `is_` meta in 2.5.0 that you'll need to incorporate. Also, I had a question about if the debug actually pushes it.

Hit me up for another review. I don't want this to hold you up.

attackant committed:
Merge branch 'release/v2.5.0' into issue-1110

attackant committed:
add screenshots

attackant committed:
rm images we dont need here

attackant committed:
Refactor post metadata retrieval in Apple News plugin for 2.5.0

The method of retrieving post metadata in the Apple News plugin has been changed to match other commits in this version.

mogmarsh reviewed (approved):
👍 looks good. thanks!
